examples/notebooks/beam-ml/rag_usecase/chunks_generation.py

```diff
@@ -0,0 +1,130 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import apache_beam as beam
+from langchain.text_splitter import CharacterTextSplitter, RecursiveCharacterTextSplitter
+from langchain_text_splitters import SentenceTransformersTokenTextSplitter
+
+from apache_beam.transforms import DoFn
+from apache_beam.transforms import PTransform
+from enum import Enum
+
+
+__all__ = [
+    'ChunksGeneration',
+    'ChunkingStrategy'
+]
+
+class ChunkingStrategy(Enum):
+    SPLIT_BY_CHARACTER = 0
+    RECURSIVE_SPLIT_BY_CHARACTER = 1
+    SPLIT_BY_TOKENS = 2
+
+
+class ChunksGeneration(PTransform):
+    """ChunkingStrategy is a ``PTransform`` that takes a ``PCollection`` of
+    key, value tuple or 2-element array and generates different chunks for documents.
+    """
+
+    def __init__(
+            self,
+            chunk_size: int,
+            chunk_overlap: int,
+            chunking_strategy: str,
```

Comment:
You can type hint the enum directly so users _have_ to use one of the provided options

https://www.geeksforgeeks.org/type-hint-enum-in-python/

Reply:
Same feedback

Reply:
@jrmccluskey My bad, I had already made the changes but missed adding this file while pushing the changes. I push the change now.